### DISCUSSION TITLE
GSTile.DemolishTile() doesn't demolish some entities

### DISCUSSION THREAD

**Discussion by andythenorth** [OpenTTD/OpenTTD#11707] [Features]
### Version of OpenTTD

~13 approx

### Expected result

The comment for GSTile.DemolishTile() is `Destroy everything on the given tile.`

Per https://docs.openttd.org/gs-api/classGSTile.html#a5eb630e5a2cc7bcb352c76c4d819cb40

Expected result is that everything on the tile is cleared.

### Actual result

Tested in both deity mode and company mode, at least the following are not demolished:

- all industries
- houses which appear to have a protected building flag set (needs confirmed)
- at least some objects, including default transmitters (needs confirmed if these have a protected flag set)
- there may be others, the ones above are what I found

Either:
- the comment is wrong and needs updated
-  or the method should demolish the items above (preferred).  Possibly an extra optional parameter might be required, to allow GS to respect items intended to be unremovable, or demolish the entire tile.

I can provide a test GS to repro if needed.

### Steps to reproduce

Write a GS that tries to clear tiles containing the entities listed above.
Observe that those entities are not cleared.

**Comment by James103** [comment]
Does the function work as intended if the magic bulldozer cheat is enabled?

**Comment by andythenorth** [comment]
> Does the function work as intended if the magic bulldozer cheat is enabled?

Yes, at least for company.  Haven't got a test immediately to hand with deity.

**Comment by mrmbernardi** [comment]
Can you provide a test GS?

**Comment by mrmbernardi** [comment]
I'd like to work on this. What's the best way to handle it? One possibility would be allowing destruction of industries/objects via the deity owner (like the water owner) but this could be a big change. I'm not sure how to add flags to the command from the script, and even if I were, it seems there's only one bit left for flags.

**Comment by PeterN** [comment]
Comment is correct but lacking context.
Basically the context is demolish a tile vs remove bits from a tile.
i.e. you can remove individual road pieces, or demolish the whole tile.
It should definitely not be expected that it will remove otherwise unremovable things.
